src/arrayMethodSort.js

```diff
@@ -4,8 +4,43 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    let len = this.length;
+    let swapped;
+
+    do {
+      swapped = false;
+
+      for (let i = 0; i < len - 1; i++) {
+        if (compareFunction) {
+          if (compareFunction(this[i], this[i + 1]) > 0) {
+            const temp = this[i];
+
+            this[i] = this[i + 1];
+            this[i + 1] = temp;
+
+            swapped = true;
+          } else if (
+            compareFunction(this[i], this[i + 1]) < 0 ||
+            compareFunction(this[i], this[i + 1]).isNan()
```

Comment:
The method `isNan()` does not exist on numbers or the result of the `compareFunction`. You probably meant to use `isNaN()` which is a global function to check if a value is NaN. However, this check is unnecessary here because if `compareFunction(this[i], this[i + 1])` is less than 0, you don't need to swap the elements, so the `else if` block can be removed.